### ISSUE TITLE
Unresolved attribute reference `StringField` for class MongoEngine

### ISSUE THREAD

**Issue by farhatnawaz** [MongoEngine/flask-mongoengine#322]
I have to write an API in Flask with MongoDB at the backend. I decided to use mongoengine for it but I'm getting attribute errors. One says `Unresolved attribute reference 'StringField' for class MongoEngine`. And this is not just for the `StringField`. I am getting the same error for all the field types I have used to far. Take a look at this screenshot:

![screen shot 2017-09-28 at 11 27 01 am](https://user-images.githubusercontent.com/8686962/30952139-0f18363e-a440-11e7-8aea-e86b3512f37e.png)

See those highlighted fields-attribute names?? Those are highlighted because those are unresolved reference errors. Any idea as to what's wrong?

P.S. Before writing the actual API, I wrote a test API with `Books` class. I was having the same error there as well but when i used to run the app and hit the respective API endpoint, the data would go into the DB and I was able to retrieve it from the DB as well. Now that's really creepy. This, then, must be some kind of warning or something? I don't know. Any ideas?

**Comment by ionin-code** [comment]
Did you solve this issue?

**Comment by dolby14** [comment]
did you solve this issue and if yes so how. Pls help me with the same issue

**Comment by aphexlog** [comment]
@farhatnawaz Any update on this? What was your resolution?

**Comment by nimit-ddprop** [comment]
https://github.com/MongoEngine/flask-mongoengine/blob/a0b0cf80ec321d29baca743e4075101db6be7d68/flask_mongoengine/__init__.py#L72-L85

Maybe this is reason. For me I think I will ditch from `flask-mongoengine` and use `mongoengine` directly with `flask`.

**Comment by ethaizone** [comment]
https://github.com/MongoEngine/flask-mongoengine/blob/a0b0cf80ec321d29baca743e4075101db6be7d68/flask_mongoengine/__init__.py#L72-L85

Maybe this is reason. For me I think I will ditch from `flask-mongoengine` and use `mongoengine` directly with `flask`.

**Comment by insspb** [comment]
@ethaizone I am agree with this comment. As current maintainer I am looking at this and thinking how to drop it completely. 
Because such silent modification looks very unpythonic for me. 
At least typechecking will always fail with it.

**Comment by insspb** [comment]
But changing this will be breaking change for all related projects.

**Comment by GonerGenesis** [comment]
any news on that?

**Comment by bestcondition** [comment]
See [Mapping Documents](https://flask.palletsprojects.com/en/2.0.x/patterns/mongoengine/#mapping-documents)

It works for me. But I don't know why

```python
import mongoengine as me

class User(me.Document):
    name = me.StringField()
    email = me.StringField()
```
![图片](https://user-images.githubusercontent.com/38112075/155872814-6c5a34ba-b8ff-49f2-9086-d415baba89c7.png)

**Comment by insspb** [comment]
@bestcondition 

It partly works like that, you will face problems if you try to use it in WTF forms with filters/validators.

@GonerGenesis, @ethaizone, @aphexlog @dolby14, @madoka-wizard @farhatnawaz 

In general I just merged #472 to master. It should fix any attribute missing highlight in IDEs. 

I completely dropped old 'silent' patching code and now everything either declared explicitly, either bypassed to original mongoengine library without modification.

At this point the problem only in dynamic docstring generation with respect to child/parent classes. I will thing something on it, but it does not affect workflow itself.


If you have some time... After 5 years of this issue. Take a look on master version. There will be some more fixes there, as project currently need complete refresh.
